src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -83,5 +124,121 @@ AS
     COMMIT TRANSACTION
 GO
 
-COMMIT TRANSACTION
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     GetExtendedQueryTagErrors
+--
+-- DESCRIPTION
+--     Gets the extended query tag errors by tag path.
+--
+-- PARAMETERS
+--     @tagPath
+--         * The TagPath for the extended query tag for which we retrieve error(s).
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.GetExtendedQueryTagErrors (@tagPath VARCHAR(64))
+AS
+	SET NOCOUNT     ON
+	SET XACT_ABORT  ON
+	BEGIN TRANSACTION
+
+
+    DECLARE @tagKey INT        
+
+	SELECT @tagKey = TagKey
+	FROM dbo.ExtendedQueryTag WITH(XLOCK)
```

Comment:
I am unsure about this `XLOCK` here. I know we use this for deleting extended query tags, but is it more appropriate here than a `HOLDLOCK` for shared locks?

On a somewhat related question: do we care about catching this case? Is there a different to the user between no errors and a non-existent query tag? Aren't they both 404s at the end of the day? If so, we could simply join the tables and return the results without any of this business.

Reply:
if the extended query tag exists and has no errors, it should mean that it was processed correctly. this would return 200
No extended query tag means the request is wrong so 404.
Am I getting what you mean correctly?